File: assp/elem/04_summation.py
""" Elementary 4:
    Write a program that asks the user for a number
    n and prints the sum of the numbers 1+2+..+n
"""

def sum_over_n(N):
    return sum(range(1,N+1))

N = int(input("Enter a positive integer: "))

assert N > 0, "The entered value must be positive"

print(f"The sum of natural numbers from 1 to {N}",
      f"is {sum_over_n(N)}")

File: assp/elem/04_summation.c
// Elementary 4: Summation of numbers from 1 to N

#include <stdio.h>

int main(void)
{
    unsigned long long int N, sum=0;

    printf("Enter a positive integer: ");
    scanf("%llu", &N);

    for (int i = 1; i <= N; i++)
    {
        sum += i;
    }

    printf("The sum of natural numbers from 1 to %llu", N);
    printf(" is %llu\n", sum);

    return 0;
}

